### PR TITLE
JDBC - Appender for Decimal

### PR DESCRIPTION
**Pull request by Jens-H**
This PR adds the appendBigDecimal method to directly insert BigDecimal into a Decimal column.

Everything down to JNI is touched, but basically the same conversion logic (BigDecimal -> Value::DECIMAL) from SQL insert is used.

A test is included.

### CODE REVIEW

File: tools/jdbc/src/test/java/org/duckdb/test/TestDuckDBJDBC.java [hunk 1]
```diff
@@ -2318,6 +2318,154 @@ public static void test_appender_date_and_time() throws Exception {
...
+
+    public static void test_appender_decimal() throws Exception {
+        DuckDBConnection conn = DriverManager.getConnection("jdbc:duckdb:").unwrap(DuckDBConnection.class);
+        Statement stmt = conn.createStatement();
+
+        stmt.execute(
+            "CREATE TABLE decimals (id INT4, a DECIMAL(4,2), b DECIMAL(8,4), c DECIMAL(18,6), d DECIMAL(38,20))");
+        DuckDBAppender appender = conn.createAppender(DuckDBConnection.DEFAULT_SCHEMA, "decimals");
+
+        BigDecimal bd16 = new BigDecimal("12.34").setScale(2);
```
Comment (Mause) [inline tools/jdbc/src/test/java/org/duckdb/test/TestDuckDBJDBC.java:2330]: Can you break this test up a bit? And use slightly more comprehensible variable names?

**Review by Mause — CHANGES_REQUESTED** [review]

File: tools/jdbc/src/jni/duckdb_java.cpp [hunk 2]
```diff
@@ -964,6 +964,38 @@ void _duckdb_jdbc_appender_append_timestamp(JNIEnv *env, jclass, jobject appende
 	get_appender(env, appender_ref_buf)->Append(Value::TIMESTAMP(timestamp));
 }
 
+void _duckdb_jdbc_appender_append_decimal(JNIEnv *env, jclass, jobject appender_ref_buf, jobject value) {
+	jint precision = env->CallIntMethod(value, J_Decimal_precision);
```
[the remark this answers is not in the document]
Comment (Jens-H) [reply tools/jdbc/src/jni/duckdb_java.cpp]: yeah, good idea
see https://github.com/duckdb/duckdb/pull/9568/commits/cadca20efee821a661816f29f2c03622fff5e16c

File: tools/jdbc/src/test/java/org/duckdb/test/TestDuckDBJDBC.java [hunk 3]
```diff
@@ -2318,6 +2318,154 @@ public static void test_appender_date_and_time() throws Exception {
...
+
+        appender = conn.createAppender(DuckDBConnection.DEFAULT_SCHEMA, "decimals");
+        appender.beginRow();
+        appender.append(6);
+        boolean exceptionRaised = false;
+
+        try {
+            appender.appendBigDecimal(new BigDecimal("121.14").setScale(2));
+        } catch (SQLException e) {
+            exceptionRaised = true;
```
[the remark this answers is not in the document]
Comment (Jens-H) [reply tools/jdbc/src/test/java/org/duckdb/test/TestDuckDBJDBC.java]: see https://github.com/duckdb/duckdb/pull/9568/commits/82ae1ac7b744c708bc0f1be1a60cf5ce6f56414a

**Comment by Jens-H** [comment]
@Mause: How to proceed here?

**Comment by Mytherin** [comment]
@Mause can this be merged?

**Review by Mause — APPROVED** [review]

**Pull request merged**